File: RealEstateAgencyC++/RealEstateAgency/Agency.cpp
#include "Agency.h"
#include <iostream>
#include <string>

using namespace std;


int Agency::apartmentCount = 0;
int Agency::houseCount = 0;
int Agency::parcelCount = 0;



Agency::Agency() {
	for (size_t i = 0; i < 5; i++)
	{
		apartments[i] = Apartment();
		houses[i] = House();
		parcels[i] = Parcel();
	}
	
	
}
Agency::~Agency() {
	delete[] apartments;
	delete[] houses;
	delete[] parcels;
}

int Agency::AddApartment(Apartment& a) {
	apartments[apartmentCount++] = a;
	return 1;
}
int Agency::AddHouse(House& h) {
	houses[houseCount++] = h;
	return 1;
}

int Agency::AddParcel(Parcel& p) {
	parcels[parcelCount++] = p;
	return 1;
}


int Agency::RemoveProperty(const char* anAddress) {
	for (size_t i = 0; i < 5; i++)
	{
		if (strcmp(apartments[i].GetAddress(), anAddress) == 0)
		{
			apartments[i] = Apartment();
			return 1;
		}

		if (strcmp(houses[i].GetAddress(), anAddress) == 0)
		{
			houses[i] = House();
			return 1;
		}

		if (strcmp(parcels[i].GetAddress(), anAddress) == 0)
		{
			parcels[i] = Parcel();
			return 1;
		}
	}
	return 0;

}

void Agency::FindByPrice(double aPrice) {
	for (size_t i = 0; i < 5; i++)
	{
		if (apartments[i].GetPrice() == aPrice)
		{
			cout << "The property with a price of " << aPrice << " is: " << endl;
			apartments[i].PropertyPrint();
		}

		if (houses[i].GetPrice() == aPrice)
		{
			cout << "The property with a price of " << aPrice << " is: " << endl;
			houses[i].PropertyPrint();
		}

		if (parcels[i].GetPrice() == aPrice)
		{
			cout << "The property with a price of " << aPrice << " is: " << endl;
			parcels[i].PropertyPrint();
		}
	}
	cout << endl;
}

void Agency::PrintProperties(const char* typeOfProperty) {
	char apartment[] = "Apartments";
	char house[] = "Houses";
	char parcel[] = "Parcels";
	Apartment tempA;
	House tempH;
	Parcel tempP;

	if (strcmp(apartment, typeOfProperty) == 0)
	{
		for (size_t i = 0; i < 5; i++)
		{
			for (size_t i = 0; i < 4; i++)
			{
				if (apartments[i].GetPrice() > apartments[i + 1].GetPrice())
				{
					tempA = apartments[i];
					apartments[i] = apartments[i + 1];
					apartments[i + 1] = tempA;
				}
			}
		}
		for (size_t i = 0; i < 5; i++)
		{
			apartments[i].PropertyPrint();

		}
	}

	if (strcmp(house, typeOfProperty) == 0)
	{
		for (size_t i = 0; i < 5; i++)
		{
			for (size_t i = 0; i < 4; i++)
			{
				if (houses[i].GetPrice() > houses[i + 1].GetPrice())
				{
					tempH = houses[i];
					houses[i] = houses[i + 1];
					houses[i + 1] = tempH;
				}
			}
		}
		for (size_t i = 0; i < 5; i++)
		{
			houses[i].PropertyPrint();

		}
	}

	if (strcmp(parcel, typeOfProperty) == 0)
	{
		for (size_t i = 0; i < 5; i++)
		{
			for (size_t i = 0; i < 4; i++)
			{
				if (parcels[i].GetPrice() > parcels[i + 1].GetPrice())
				{
					tempP = parcels[i];
					parcels[i] = parcels[i + 1];
					parcels[i + 1] = tempP;
				}
			}
		}

		for (size_t i = 0; i < 5; i++)
		{
			parcels[i].PropertyPrint();

		}
	}
	
}

File: RealEstateAgencyCS/RealEstateAgencyCS/Agency.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RealEstateAgencyCS
{
    class Agency
    {
        private static int apartmentCound = 0;
        private static int houseCount = 0;
        private static int parcelCount = 0;
        private static int propertyCount;
        private static string agencyName;
        Apartment[] apartments = new Apartment[5];
        House[] houses = new House[5];
        Parcel[] parcels = new Parcel[5];

        public Agency()
        {
            for (int i = 0; i < 5; i++)
            {
                apartments[i] = new Apartment();
                houses[i] = new House();
                parcels[i] = new Parcel();
            }
        }

        public int AddApartment(Apartment a)
        {
            apartments[apartmentCound++] = a;
            return 1;
        }

        public int AddHouse(House h)
        {
            houses[houseCount++] = h;
            return 1;
        }

        public int AddParcel(Parcel p)
        {
            parcels[parcelCount++] = p;
            return 1;
        }

        public int RemoveProperty(string anAddress)
        {
            for (int i = 0; i < 5; i++)
            {
                if (apartments[i].GetAddress() == anAddress)
                {
                    apartments[i] = new Apartment();
                    return 1;
                }

                if (houses[i].GetAddress() == anAddress)
                {
                    houses[i] = new House();
                    return 1;
                }

                if (parcels[i].GetAddress() == anAddress)
                {
                    parcels[i] = new Parcel();
                    return 1;
                }
            }
            return 0;
        }

        public void FindByPrice(double aPrice)
        {
            for (int i = 0; i < 5; i++)
            {
                if (apartments[i].GetPrice() == aPrice)
                {
                    Console.WriteLine($"The property with a price of {apartments[i].GetPrice()} is: ");
                    apartments[i].PropertyPrint();
                }

                if (houses[i].GetPrice() == aPrice)
                {
                    Console.WriteLine($"The property with a price of {houses[i].GetPrice()} is: ");
                    houses[i].PropertyPrint();
                }

                if (parcels[i].GetPrice() == aPrice)
                {
                    Console.WriteLine($"The property with a price of {parcels[i].GetPrice()} is: ");
                    parcels[i].PropertyPrint();
                }
            }
            Console.WriteLine();
        }

        public void PrintProperties(string typeOfProperty)
        {
            string apartment = "Apartments";
            string house = "Houses";
            string parcel = "Parcels";
            Apartment tempA;
            House tempH;
            Parcel tempP;

            if (apartment == typeOfProperty)
            {
                for (int i = 0; i < 5; i++)
                {
                    for (int j = 0; j < 4; j++)
                    {
                        if (apartments[j].GetPrice() > apartments[j + 1].GetPrice())
                        {
                            tempA = apartments[j];
                            apartments[j] = apartments[j + 1];
                            apartments[j + 1] = tempA;
                        }
                    }
                }
                for (int i = 0; i < 5; i++)
                {
                    apartments[i].PropertyPrint();
                }
            }

            if (house == typeOfProperty)
            {
                for (int i = 0; i < 5; i++)
                {
                    for (int j = 0; j < 4; j++)
                    {
                        if (houses[j].GetPrice() > houses[j + 1].GetPrice())
                        {
                            tempH = houses[j];
                            houses[j] = houses[j + 1];
                            houses[j + 1] = tempH;
                        }
                    }
                }
                for (int i = 0; i < 5; i++)
                {
                    houses[i].PropertyPrint();
                }
            }

            if (parcel == typeOfProperty)
            {
                for (int i = 0; i < 5; i++)
                {
                    for (int j = 0; j < 4; j++)
                    {
                        if (parcels[j].GetPrice() > parcels[j + 1].GetPrice())
                        {
                            tempP = parcels[j];
                            parcels[j] = parcels[j + 1];
                            parcels[j + 1] = tempP;
                        }
                    }
                }
                for (int i = 0; i < 5; i++)
                {
                    parcels[i].PropertyPrint();
                }
            }

        }


    }
}
